### PR TITLE
Asterisk

### PR DESCRIPTION
**Pull request by jkroonza**

### CODE REVIEW

**Comment by jkroonza** [comment]
More detailed diff:

https://dpaste.com/3YC89N5G6

**Comment by jkroonza** [comment]
The last push merely stripped elogind code.  systemd integration is limited to port-activation.  There isn't a way in the configure script to force enable/disable systemd integration that I can find.  Could write it, but I'm not sure it's worth the effort.

**Comment by jkroonza** [comment]
Last push added libxml2 as BDEPEND, this is used by menuselect, which is executed on the BUILD host, and requires libxml2.

File: net-misc/asterisk/asterisk-18.2.2.ebuild [hunk 1]
```diff
@@ -0,0 +1,354 @@
...
+	systemd_install_serviced "${FILESDIR}"/asterisk.service.conf
+
+	# Reset diropts else dodoc uses it for doc installations.
+	diropts -m0755
+
+	# install the upgrade documentation
+	dodoc UPGRADE* BUGS CREDITS
+
+	# install extra documentation
+	use doc &&doc/*.{txt,pdf}
```
Comment (juippis) [inline net-misc/asterisk/asterisk-18.2.2.ebuild:309]:
```suggestion
use doc && doc/*.{txt,pdf}
```
(I don't mind this, but if you need to modify the ebuild anyway. IF.)

**Review by juippis — COMMENTED** [review]
So here's what I'm seeing, on multiple tests, all versions:
```
Updating asterisk.conf
/var/tmp/portage/net-misc/asterisk-18.2.2/temp/environment: line 2842: doc/IAX2-security.txt:       Permission denied
>>> Completed installing net-misc/asterisk-18.2.2 into /var/tmp/portage/net-misc/asterisk-18.2.2/   image

 * Final size of build directory: 131804 KiB (128.7 MiB)
 * Final size of installed tree:   23036 KiB ( 22.4 MiB)

 * The ebuild is installing to one or more unexpected paths:
 *
 *   /tmp
 *
 * Please fix the ebuild to use correct FHS/Gentoo policy paths.
```
Although the permission denied won't happen everytime, the `/tmp` seems to be occurring eveytime.

**Comment by juippis** [comment]
```
 # grep -r "Permission denied" /var/tmp/portage/vbslogs
/var/tmp/portage/vbslogs/build/net-misc/asterisk-16.16.2:20210317-070323.log:/var/tmp/portage/net-misc/asterisk-16.16.2/temp/environment: line 2808: doc/IAX2-security.txt: Permission denied
/var/tmp/portage/vbslogs/build/net-misc/asterisk-13.38.2-r1:20210317-071334.log:/var/tmp/portage/net-misc/asterisk-13.38.2-r1/temp/environment: line 2823: doc/IAX2-security.txt: Permission denied
/var/tmp/portage/vbslogs/build/net-misc/asterisk-16.16.2-r1:20210317-071854.log:/var/tmp/portage/net-misc/asterisk-16.16.2-r1/temp/environment: line 2842: doc/IAX2-security.txt: Permission denied
/var/tmp/portage/vbslogs/build/net-misc/asterisk-16.16.2:20210317-065400.log:/var/tmp/portage/net-misc/asterisk-16.16.2/temp/environment: line 2808: doc/IAX2-security.txt: Permission denied
/var/tmp/portage/vbslogs/build/net-misc/asterisk-18.2.2:20210317-072021.log:/var/tmp/portage/net-misc/asterisk-18.2.2/temp/environment: line 2839: doc/IAX2-security.txt: Permission denied
/var/tmp/portage/vbslogs/build/net-misc/asterisk-13.38.2:20210317-070748.log:/var/tmp/portage/net-misc/asterisk-13.38.2/temp/environment: line 2800: doc/IAX2-security.txt: Permission denied
/var/tmp/portage/vbslogs/build/net-misc/asterisk-18.2.2:20210317-072148.log:/var/tmp/portage/net-misc/asterisk-18.2.2/temp/environment: line 2842: doc/IAX2-security.txt: Permission denied
/var/tmp/portage/vbslogs/build/net-misc/asterisk-16.16.2-r1:20210317-071614.log:/var/tmp/portage/net-misc/asterisk-16.16.2-r1/temp/environment: line 2839: doc/IAX2-security.txt: Permission denied
```

```
# grep -r "installing to one or more unexpected paths" /var/tmp/portage/vbslogs
/var/tmp/portage/vbslogs/build/net-misc/asterisk-16.16.2:20210317-070153.log: * The ebuild is installing to one or more unexpected paths:
/var/tmp/portage/vbslogs/build/net-misc/asterisk-16.16.2:20210317-070323.log: * The ebuild is installing to one or more unexpected paths:
/var/tmp/portage/vbslogs/build/net-misc/asterisk-16.16.2:20210317-065802.log: * The ebuild is installing to one or more unexpected paths:
/var/tmp/portage/vbslogs/build/net-misc/asterisk-16.16.2-r1:20210317-071854.log: * The ebuild is installing to one or more unexpected paths:
/var/tmp/portage/vbslogs/build/net-misc/asterisk-16.16.2:20210317-065400.log: * The ebuild is installing to one or more unexpected paths:
/var/tmp/portage/vbslogs/build/net-misc/asterisk-18.2.2:20210317-072021.log: * The ebuild is installing to one or more unexpected paths:
/var/tmp/portage/vbslogs/build/net-misc/asterisk-16.16.2-r1:20210317-071732.log: * The ebuild is installing to one or more unexpected paths:
/var/tmp/portage/vbslogs/build/net-misc/asterisk-18.2.2:20210317-072148.log: * The ebuild is installing to one or more unexpected paths:
/var/tmp/portage/vbslogs/build/net-misc/asterisk-16.16.2-r1:20210317-071614.log: * The ebuild is installing to one or more unexpected paths:
/var/tmp/portage/vbslogs/elog/net-misc/asterisk-16.16.2:20210317-065518.log:The ebuild is installing to one or more unexpected paths:
/var/tmp/portage/vbslogs/elog/net-misc/asterisk-16.16.2-r1:20210317-071847.log:The ebuild is installing to one or more unexpected paths:
/var/tmp/portage/vbslogs/elog/net-misc/asterisk-16.16.2-r1:20210317-071723.log:The ebuild is installing to one or more unexpected paths:
/var/tmp/portage/vbslogs/elog/net-misc/asterisk-18.2.2:20210317-072258.log:The ebuild is installing to one or more unexpected paths:
/var/tmp/portage/vbslogs/elog/net-misc/asterisk-18.2.2:20210317-072137.log:The ebuild is installing to one or more unexpected paths:
/var/tmp/portage/vbslogs/elog/net-misc/asterisk-16.16.2-r1:20210317-072008.log:The ebuild is installing to one or more unexpected paths:
/var/tmp/portage/vbslogs/elog/net-misc/asterisk-16.16.2:20210317-070444.log:The ebuild is installing to one or more unexpected paths:
/var/tmp/portage/vbslogs/elog/net-misc/asterisk-16.16.2:20210317-065923.log:The ebuild is installing to one or more unexpected paths:
/var/tmp/portage/vbslogs/elog/net-misc/asterisk-16.16.2:20210317-070319.log:The ebuild is installing to one or more unexpected paths:
```

**Comment by jkroonza** [comment]
https://dpaste.com/GB6ACYFRZ

Shows diffs for 13 and 16 ebuilds (in branch), and again for 16 => 18.

Also shown diffs for updated confd and initd.

Edited no-var-run-install.patch in place since it doesn't affect existing 16 installs (will prevent install of /tmp for those, but it doesn't affect existing installations).

Quite a bit of work to rather do media caches in /var/cache/asterisk where it belongs (but like /tmp we shouldn't install in there ... which is the reason for updated tmpfiles and edit of initd and confd).

**Review by juippis — APPROVED** [review]

**Comment by juippis** [comment]
All good now, thanks a lot!

**Pull request closed without merging**